### ISSUE TITLE
why minus maximum in function entropy in a2c ?

### ISSUE THREAD

**Issue by Griffithchiong** [openai/baselines#761]
a_0 = self.logits - tf.reduce_max(self.logits, axis=-1, keepdims=True)
I got confused for a very long time, any information is welcome, thanks a lot!

**Comment by DanielTakeshi** [comment]
It's probably due to numerical instability with exponentiating large numbers. You can remove a constant from each value and still get the same semantics from the softmax.

**Comment by Griffithchiong** [comment]
> It's probably due to numerical instability with exponentiating large numbers. You can remove a constant from each value and still get the same semantics from the softmax.
yes, you are absolutely right!!
I tried this:

with tf.Session() as sess:
    r = tf.exp(300.0)/tf.exp(301.0)
    print(sess.run(r))
and got an Nan return, 
when I check the tensorflow how to do with the softmax function, it's too complicated for me, maybe I will spend more time later
Thank you so much!!
I got confused for a very long time!!
